src/core/playground/accountAuthenticator/singleKey.ts

```diff
@@ -0,0 +1,31 @@
+import type { Deserializer, Serializer } from "../../../bcs";
+import { HexInput } from "../../../types";
+import type { PublicKey } from "../interfaces";
+import { type AllowedSignatures, AnyPublicKey, AnySignature } from "../wrapped";
+import { AccountAuthenticatorVariant } from "./variant";
+
+export class AccountAuthenticatorSingleKey<
+  TSignature extends AllowedSignatures = AllowedSignatures,
+  TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+> {
+  constructor(
+    public readonly publicKey: AnyPublicKey<TSignature, TPublicKey>,
```

Comment:
the format we use in sdk is define the class properties and then set it in the constructor
```
class AccountAuthenticatorSingleKey<>{
 public readonly publicKey: AnyPublicKey<TSignature, TPublicKey>;
 public readonly signature: AnySignature<TSignature>

 constructor(){
    ......
 }
}
```